File: api/src/main/java/net/blueshell/api/dto/survey/AnswerDTO.java
package net.blueshell.api.dto.survey;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.EqualsAndHashCode;
import net.blueshell.api.base.BaseDTO;

import java.util.List;

@Data
@EqualsAndHashCode(callSuper = true)
@Schema(name = "Answer")
public class AnswerDTO extends BaseDTO {
    private Long id;

    @NotNull
    private Long questionId;

    private List<Boolean> optionSelections;

    private String textResponse;
}

File: api/src/main/kotlin/net/blueshell/api/dto/survey/AnswerDTO.kt
package net.blueshell.api.dto.survey

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.NotNull
import net.blueshell.api.base.BaseDTO
import net.blueshell.api.validation.survey.ValidAnswer
@Schema(name = "Answer")
@ValidAnswer
class AnswerDTO : BaseDTO() {
    @NotNull
    val questionId: @NotNull Long? = null

    val optionSelections: MutableList<Boolean?>? = null

    val textResponse: String? = null
}
